### PR TITLE
Faucet page style optimization

### PR DESCRIPTION
**Pull request by IronLu233**
This PR Optimize the UI of faucet, and refactor many code.

https://github.com/Magickbase/ckb-testnet-faucet/assets/20639676/c7cadecc-10c3-48b0-9f7a-4f1ad5a7c701

**DO NOT MERGE IT WITHOUT QA TESTING!**, it is a large PR.

### CODE REVIEW

**Comment by Keith-CY** [comment]
> DO NOT MERGE IT WITHOUT QA TESTING!, it is a large PR.

It can be changed into a draft PR

File: app/javascript/hooks/useDataAPI.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  "officialAccount" | "userAccount" | "claimEvents"
+>;
+type ClaimEventResponse = APIResponse["claimEvents"];
+
+export function useClaimEvents(specificAddress?: string) {
+  let { data, ...rest } = useSWR(
+    `/claim_events/${specificAddress || ""}`,
+    (url) => axios.get<ClaimEventResponse | APIResponse>(url),
+    {
+      refreshInterval: 1000,
```
Comment (Keith-CY) [inline app/javascript/hooks/useDataAPI.ts:15]: The request can be disabled if `address` is invalid, e.g. `undefined`

File: app/javascript/components/ClaimEventForm.tsx [hunk 2]
```diff
@@ -1,109 +1,184 @@
...
 
-      </Form.Group>
+  const validationSchema = Yup.object().shape({
+    addressHash: Yup.string()
+      .trim()
+      .min(40, "Invalid address")
+      .matches(/ckt/, "Invalid address")
+      .test(
+        "remaining",
+        "The amount you claimed are greater than your remaining.",
```
Comment (Keith-CY) [inline app/javascript/components/ClaimEventForm.tsx]:
```suggestion
        "The amount you claimed cannot be greater than your remaining.",
```

File: app/javascript/components/ClaimEventForm.tsx [hunk 3]
```diff
@@ -1,109 +1,184 @@
...
+  const { mutate: freshClaimEvents } = useClaimEvents();
+  const addressHashRef = useRef<string>();
+  const remainingRef = useRef<number | null | undefined>();
 
-      </Form.Group>
+  const validationSchema = Yup.object().shape({
+    addressHash: Yup.string()
+      .trim()
+      .min(40, "Invalid address")
+      .matches(/ckt/, "Invalid address")
```
Comment (Keith-CY) [inline app/javascript/components/ClaimEventForm.tsx:29]: Address validation should be done with sdk

Comment (IronLu233) [reply app/javascript/components/ClaimEventForm.tsx:29]: OK, I will add it to make it more precision.

Comment (IronLu233) [reply app/javascript/components/ClaimEventForm.tsx:29]: I'm sorry to tell a technical limitation here.

When I use [@ckb-lumos](https://www.npmjs.com/package/@ckb-lumos/lumos?activeTab=readme) to implement it, the page can not be rendered by Ruby on rails. although the frontend code seems to be compiled successful(I confirmed the cache have been cleared), but Ruby on Rails still throw an 500 error. I can not get extra information regarding this error anymore.

The error log is only here.
```sh
[Webpacker] Compiling...
[Webpacker] Compiled all packs in /home/ubuntu/ckb-testnet-faucet/public/packs
[Webpacker] Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating

[Webpacker] Hash: 86f3f49b6fa2cc784568
Version: webpack 4.41.6
Time: 6124ms
Built at: 06/06/2023 2:13:50 PM
                                                              Asset       Size            Chunks                         Chunk Names
                             js/application-05ff4e2a3884299dc9f1.js   5.32 MiB       application  [emitted] [immutable]  application
                         js/application-05ff4e2a3884299dc9f1.js.map   5.23 MiB       application  [emitted] [dev]        application
                        js/server_rendering-b1d3b581262b4aed157b.js   4.58 MiB  server_rendering  [emitted] [immutable]  server_rendering
                    js/server_rendering-b1d3b581262b4aed157b.js.map   4.47 MiB  server_rendering  [emitted] [dev]        server_rendering
                                                      manifest.json   2.28 KiB                    [emitted]              
               media/images/bg-c09bcc8975562d0a28534a51a971af41.jpg    268 KiB                    [emitted]              
            media/images/ckb-n-72dbf1b4ed1a034e7019884c304fec78.png   16.1 KiB                    [emitted]              
         media/images/ckb_logo-30bc8adb5ebf507e49752f34c68a9f31.png   5.26 KiB                    [emitted]              
            media/images/clear-757d57e113739e4e933307a463dabc5f.svg   2.39 KiB                    [emitted]              
     media/images/footer_about-5f65f295f8a4fd19b91917a9d7ce3901.png   1.19 KiB                    [emitted]              
      media/images/footer_blog-1d6b39333ea0c73b97266b017705d27b.png  759 bytes                    [emitted]              
      media/images/footer_docs-1d7d3fdfb86b7651b11e9e5cb569b18c.png  443 bytes                    [emitted]              
     media/images/footer_forum-90be852d65b7bb2993e5cb80c4a0f621.png  867 bytes                    [emitted]              
    media/images/footer_github-080554f388b33089654078bba4b74dd8.png   1.46 KiB                    [emitted]              
    media/images/footer_reddit-493bbcd1e7446e03e6eb6758f497f704.png   1.99 KiB                    [emitted]              
  media/images/footer_telegram-1ee7bafa684de61f8cbda5b400664ca9.png   1.92 KiB                    [emitted]              
   media/images/footer_twitter-96a1542d436cf8a03efe353cb1bf28b4.png   1.64 KiB                    [emitted]              
media/images/footer_whitepaper-9bdf13d2627c4b53558cc93004d82401.png  835 bytes                    [emitted]              
   media/images/footer_youtube-107d36d1624e4fbd4f4e3543c259695f.png  769 bytes                    [emitted]              
  media/images/mobile_ckb_logo-30bc8adb5ebf507e49752f34c68a9f31.png   5.26 KiB                    [emitted]              
Entrypoint application = js/application-05ff4e2a3884299dc9f1.js js/application-05ff4e2a3884299dc9f1.js.map
Entrypoint server_rendering = js/server_rendering-b1d3b581262b4aed157b.js js/server_rendering-b1d3b581262b4aed157b.js.map
[./app/javascript/channels sync recursive _channel\.js$] ./app/javascript/channels sync _channel\.js$ 160 bytes {application} [built]
[./app/javascript/channels/index.js] 211 bytes {application} [built]
[./app/javascript/components sync recursive ^\.\/.*$] ./app/javascript/components sync ^\.\/.*$ 382 bytes {application} {server_rendering} [built]
[./app/javascript/hooks/useDataAPI.ts] 2.46 KiB {application} {server_rendering} [built]
[./app/javascript/images/bg.jpg] 98 bytes {application} [built]
[./app/javascript/images/ckb-n.png] 101 bytes {application} {server_rendering} [built]
[./app/javascript/images/ckb_logo.png] 104 bytes {application} [built]
[./app/javascript/images/clear.svg] 101 bytes {application} [built]
[./app/javascript/images/footer_about.png] 108 bytes {application} {server_rendering} [built]
[./app/javascript/images/footer_blog.png] 107 bytes {application} {server_rendering} [built]
[./app/javascript/packs/application.js] 1010 bytes {application} [built]
[./app/javascript/packs/server_rendering.js] 301 bytes {server_rendering} [built]
[./app/javascript/stylesheets/application.scss] 706 bytes {application} [built]
[./node_modules/css-loader/dist/cjs.js?!./node_modules/postcss-loader/src/index.js?!./node_modules/sass-loader/dist/cjs.js?!./app/javascript/stylesheets/application.scss] ./node_modules/css-loader/dist/cjs.js??ref--8-1!./node_modules/postcss-loader/src??ref--8-2!./node_modules/sass-loader/dist/cjs.js??ref--8-3!./app/javascript/stylesheets/application.scss 622 KiB {application} [built]
[./node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 552 bytes {application} {server_rendering} [built]
    + 460 hidden modules

[Webpacker] Everything's up-to-date. Nothing to do
Completed 500 Internal Server Error in 6807ms (Views: 0.1ms | ActiveRecord: 2.5ms | Allocations: 32390)

```

**Pull request closed without merging**